# Covariance & Correlation

In [2]:
#import data set using pandas 
import pandas as pd
dataset= pd.read_csv("PrePlacement.csv")

In [3]:
dataset

sl_no gender  ssc_p    ssc_b  hsc_p    hsc_b     hsc_s  degree_p  \
0        1      M  67.00   Others  91.00   Others  Commerce     58.00   
1        2      M  79.33  Central  78.33   Others   Science     77.48   
2        3      M  65.00  Central  68.00  Central      Arts     64.00   
3        4      M  56.00  Central  52.00  Central   Science     52.00   
4        5      M  85.80  Central  73.60  Central  Commerce     73.30   
..     ...    ...    ...      ...    ...      ...       ...       ...   
210    211      M  80.60   Others  82.00   Others  Commerce     77.60   
211    212      M  58.00   Others  60.00   Others   Science     72.00   
212    213      M  67.00   Others  67.00   Others  Commerce     73.00   
213    214      F  74.00   Others  66.00   Others  Commerce     58.00   
214    215      M  62.00  Central  58.00   Others   Science     53.00   

      degree_t workex  etest_p specialisation  mba_p      status    salary  
0     Sci&Tech     No     55.0         Mkt&HR  58.80      Placed  270000.0  
1     Sci&Tech    Yes     86.5        Mkt&Fin  66.28      Placed  200000.0  
2    Comm&Mgmt     No     75.0        Mkt&Fin  57.80      Placed  250000.0  
3     Sci&Tech     No     66.0         Mkt&HR  59.43  Not Placed       0.0  
4    Comm&Mgmt     No     96.8        Mkt&Fin  55.50      Placed  425000.0  
..         ...    ...      ...            ...    ...         ...       ...  
210  Comm&Mgmt     No     91.0        Mkt&Fin  74.49      Placed  400000.0  
211   Sci&Tech     No     74.0        Mkt&Fin  53.62      Placed  275000.0  
212  Comm&Mgmt    Yes     59.0        Mkt&Fin  69.72      Placed  295000.0  
213  Comm&Mgmt     No     70.0         Mkt&HR  60.23      Placed  204000.0  
214  Comm&Mgmt     No     89.0         Mkt&HR  60.22  Not Placed       0.0  

[215 rows x 15 columns]

In [4]:
dataset.isna().sum() #To check the null values

sl_no             0
gender            0
ssc_p             0
ssc_b             0
hsc_p             0
hsc_b             0
hsc_s             0
degree_p          0
degree_t          0
workex            0
etest_p           0
specialisation    0
mba_p             0
status            0
salary            0
dtype: int64

In [5]:
#To get the covariance
dataset.cov()

sl_no          ssc_p          hsc_p       degree_p  \
sl_no      3870.000000     -52.641355     -59.598879     -41.465047   
ssc_p       -52.641355     117.228377      58.853253      42.702550   
hsc_p       -59.598879      58.853253     112.063731      33.684453   
degree_p    -41.465047      42.702550      33.684453      53.604710   
etest_p      52.556168      37.659225      33.838355      22.078774   
mba_p         8.102336      24.535952      21.517688      17.185200   
salary    11383.177570  908858.485818  731007.850848  466336.264888   

                etest_p          mba_p        salary  
sl_no         52.556168       8.102336  1.138318e+04  
ssc_p         37.659225      24.535952  9.088585e+05  
hsc_p         33.838355      21.517688  7.310079e+05  
degree_p      22.078774      17.185200  4.663363e+05  
etest_p      176.251018      16.886973  3.727004e+05  
mba_p         16.886973      34.028376  1.239934e+05  
salary    372700.449468  123993.387361  2.259185e+10

#The score 22.078 is a positive covariance between degree_p vs etest_p
#The score 16.078 is a positive covariance between etest_p vs mba_p


In [17]:
#To get the correlation
dataset.corr()

sl_no     ssc_p     hsc_p  degree_p   etest_p     mba_p    salary
sl_no     1.000000 -0.078155 -0.090500 -0.091039  0.063636  0.022327  0.001217
ssc_p    -0.078155  1.000000  0.513478  0.538686  0.261993  0.388478  0.558475
hsc_p    -0.090500  0.513478  1.000000  0.434606  0.240775  0.348452  0.459424
degree_p -0.091039  0.538686  0.434606  1.000000  0.227147  0.402376  0.423762
etest_p   0.063636  0.261993  0.240775  0.227147  1.000000  0.218055  0.186775
mba_p     0.022327  0.388478  0.348452  0.402376  0.218055  1.000000  0.141417
salary    0.001217  0.558475  0.459424  0.423762  0.186775  0.141417  1.000000

##The score 0.14 is a positive correlation between Salary vs mba_p